http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -465,14 +465,29 @@ object clusterQuery extends TableQuery(new ClusterTable(_)) {
 
   def getStagingBucket(cloudContext: CloudContext, name: RuntimeName)(implicit
     ec: ExecutionContext
-  ): DBIO[Option[GcsPath]] =
+  ): DBIO[Option[StagingBucket]] =
     clusterQuery
       .filter(_.cloudContextDb === cloudContext.asCloudContextDb)
       .filter(_.runtimeName === name)
-      .map(_.stagingBucket)
+      .map(x => (x.cloudProvider, x.stagingBucket))
       .result
       // staging bucket is saved as a bucket name rather than a path
-      .map(recs => recs.headOption.flatten.flatMap(head => parseGcsPath("gs://" + head + "/").toOption))
+      .map(recs =>
+        recs.headOption.flatMap { head =>
+          head._1 match {
+            case CloudProvider.Gcp => head._2.map(s => StagingBucket.Gcp(GcsBucketName(s)))
+            case CloudProvider.Azure =>
+              head._2.map { s =>
+                val res = for {
+                  splitted <- Either.catchNonFatal(s.split("/"))
```

Comment:
nit: `splitted` sounds weird can we have it say `split`

Reply:
how about `splittedString` just to be more specific? I think it's fairly common to call it this way

Reply:
meh, not super important. just don't like the `ed` which makes it a verb in the past tense, and the other vars are nouns